neqo-qpack/src/decoder.rs

```diff
@@ -53,8 +54,8 @@ impl Decoder {
             send_buf,
             local_stream_id: None,
             max_table_size: qpack_settings.max_table_size_decoder,
-            max_blocked_streams: usize::from(qpack_settings.max_blocked_streams),
-            blocked_streams: Vec::new(),
+            max_blocked_streams,
+            blocked_streams: Vec::with_capacity(max_blocked_streams),
```

Comment:
Want to try SmallVec here?

Reply:
With what constant size?

Reply:
You know, zero seems about right, the more I think on it.  (Which means reverting the change.)  Blocking should be avoided in most cases, so starting with no allocation is best.  If this ever needs to be allocated, paying a tax then is reasonable.

Reply:
Oh, with one potential tweak: We could allocate the maximum if we ever need to allocate.  I think that you can use `Vec::reserve()` for that.

Reply:
I don't think that's what `reserve` does: https://users.rust-lang.org/t/is-vec-with-capacity-like-vec-new-with-vec-reserve-or-vec-new-with-vec-reserve-exact/80282